projects/driver_controls/inc/cruise.h

```diff
@@ -1,44 +1,42 @@
 #pragma once
 // Cruise control target speed module
 // Requires CAN to be initialized
-// Stores updates to target cruise speed, reads motor speed from the motor controllers
-// Set the source to motor controller to record the current speed when driving normally.
-// Set the source to stored value to start cruise with the current speed and offset it to change the
-// target speed.
+// Stores updates to target cruise speed and reads motor speed from the motor controllers.
+// On events:
+// * INPUT_EVENT_CONTROL_STALK_DIGITAL_CC_SET_PRESSED: Sets the target speed to the current speed
+// * INPUT_EVENT_CONTROL_STALK_ANALOG_CC_SPEED_PLUS: Offsets the target speed by +CRUISE_OFFSET_CMS
+// * INPUT_EVENT_CONTROL_STALK_ANALOG_CC_SPEED_MINUS: Offsets the target speed by -CRUISE_OFFSET_CMS
 #include <stdbool.h>
 #include <stdint.h>
 #include "event_queue.h"
+#include "soft_timer.h"
 #include "status.h"
 
 // How to much to increment/decrement in cm/s (m/s * 100)
 // Arbitrary default of ~1mph
 #define CRUISE_OFFSET_CMS 45
-
-typedef enum {
-  CRUISE_SOURCE_MOTOR_CONTROLLER = 0,
-  CRUISE_SOURCE_STORED_VALUE,
-  NUM_CRUISE_SOURCES
-} CruiseSource;
+// Arbitrary maximum of ~60mph
```

Comment:
Absolute max. Based on our motor torque/speed curves + wheel diameter I'd be surprised if we hit 60 MPH. I can set the cruise max to a higher value though.

Since we're primarily driving in the US I figured keeping cruise as MPH is easier to deal with (otherwise we need some way to switch), but I can add a comment mentioning the conversion.